### ISSUE TITLE
Intellisense doesn't seem to work

### ISSUE THREAD

**Issue by hbkmadness** [Microsoft/vscode-go#1524]
Ok so I installed the plugin and I have this example code

```
package main

import (
    "fmt"

    "github.com/hyperledger/fabric/core/chaincode/shim"
    "github.com/hyperledger/fabric/protos/peer"
)

// SimpleAsset implements a simple chaincode to manage an asset
type SimpleAsset struct {
}

// Init is called during chaincode instantiation to initialize any data.
func (t *SimpleAsset) Init(stub shim.ChaincodeStubInterface) peer.Response {

}
```
But the intellisense doesn't seem to work for the "shim" and "peer" imports.

I ran the `gocode close` and then `gocode -debug -s` commands and the output is the following:

```
2018/02/19 12:44:09 Import path "github.com/hyperledger/fabric/core/chaincode/shim" was not resolved
2018/02/19 12:44:09 Gocode's build context is:
2018/02/19 12:44:09  GOROOT: /usr/local/go
2018/02/19 12:44:09  GOPATH: /home/slavi/go
2018/02/19 12:44:09  GOOS: linux
2018/02/19 12:44:09  GOARCH: amd64
2018/02/19 12:44:09  BzlProjectRoot: ""
2018/02/19 12:44:09  GBProjectRoot: ""
2018/02/19 12:44:09  lib-path: ""
```

(same for peer)

Now I don't understand why it cannot resolve the path, the imports work (they are not underlined in red) in the file itself and if I change anything in the import path they become underlined which means this part of the plug-in works, but the auto-completion fails.

Does anyone have an idea what could I try?

**Comment by ljubomirb** [comment]
Try Go 1.9.4 instead of 1.10 and see if there is any difference.

**Comment by ramya-rao-a** [comment]
@ljubomirb Just curious... How do you know that @hbkmadness is using 1.10?

@hbkmadness Ensure you have the latest `gocode` by running `go get -u github.com/nsf/gocode` and try again. If it still gives you problems, then please log an issue at https://github.com/nsf/gocode

**Comment by ljubomirb** [comment]
@ramya-rao-a I'm not. I have (non updated) linux setup, and everything works, and this weekend I tried fresh Windows setup on different machine, and it does not work as expected. It's not even possible to hoover over function to see its return type for example. But I cannot put my finger on it, since I'm not sure if is it me (am I also missing some package?), or is it just go 1.10 that I have freshly installed. So before blaming, I'm trying to see if someone has similar problems that started with 1.10

**Comment by ljubomirb** [comment]
Could this be related to #1527?

**Comment by hbkmadness** [comment]
Hi,

I am really running 1.10 lol but I will be able to try 1.9.4 maybe tomorrow so it will take some time until I can confirm that indeed that is the problem.

@ramya-rao-a I'm running the latest version of gocode. And I will log an issue once I try the Go 1.9.4. Long story short I've never used Go before yesterday so everything that I installed is pretty much the latest versions of everything.

**Comment by ramya-rao-a** [comment]
@hbkmadness Does intellisense work otherwise? For example when using local variables or std library? 

For external packages, `gocode` works only if the packages are available under $GOPATH/pkgs i.e it doesn't work with source files. So one thing you can try is running the below which will install the packages to $GOPATH/pkgs
```
go install github.com/hyperledger/fabric/core/chaincode/shim
go install github.com/hyperledger/fabric/protos/peer
```

@ljubomirb If you see a similar call stack as discussed in #1527, then see https://github.com/Microsoft/vscode-go/issues/1527#issuecomment-367036064

**Comment by hbkmadness** [comment]
@ramya-rao-a This worked and solved my problem.

I don't know if this is related to the issue but before installing I had to run also `sudo apt install libltdl-dev` but I think this is related with those specific pkgs.

Thank you.

**Comment by ljubomirb** [comment]
@ramya-rao-a 
Just for fun, I've removed go 1.10, installed go 1.9.4 on the same windows machine and it all started to work as expected (when mouse was over function, first "loading" was shown, then soon function details).
Then I installed the new version of vscode-go, and a new go 1.10. Automatically, some updating happen, log showing:

> Installing github.com/nsf/gocode SUCCEEDED
Installing github.com/uudashr/gopkgs/cmd/gopkgs SUCCEEDED
Installing github.com/ramya-rao-a/go-outline SUCCEEDED
Installing github.com/acroca/go-symbols SUCCEEDED
Installing golang.org/x/tools/cmd/guru SUCCEEDED
Installing golang.org/x/tools/cmd/gorename SUCCEEDED
Installing github.com/fatih/gomodifytags SUCCEEDED
Installing github.com/haya14busa/goplay/cmd/goplay SUCCEEDED
Installing github.com/josharian/impl SUCCEEDED
Installing github.com/davidrjenni/reftools/cmd/fillstruct SUCCEEDED
Installing github.com/zmb3/gogetdoc SUCCEEDED
Installing sourcegraph.com/sqs/goreturns SUCCEEDED
Installing github.com/golang/lint/golint SUCCEEDED
Installing github.com/cweill/gotests/... SUCCEEDED
Installing github.com/derekparker/delve/cmd/dlv SUCCEEDED
All tools successfully installed. You're ready to Go :).

All I can say is that it now everything works, even with Go 1.10 and I hope this little test I did somehow also helps you in future in some way.
Great product btw! :)
